### DISCUSSION TITLE
Could this be used to manage external macOS runners?

### DISCUSSION THREAD

**Discussion by codiophile** [actions/actions-runner-controller#2297] [Questions]
We are using this for our self-hosted linux runners, but we also need to build stuff on macOS. My question is simply whether this project could be extended to manage external runners and spin up and down macOS VMs in an external cluster and add them as runners to the appropriate repository? If so, how would one go about adding an integration with an external cluster?

Since this project already implements webhooks and auto-scaling, it seems like it could potentially be a good fit and we already use it anyway. Thanks.

**Comment by hibby96** [comment]
@codiophile  I'm looking to do exactly what you described.  Did you come up with or find any good solutions for this?
